### PR TITLE
dotCMS/core#24236 When Experiment is running Variant modifications should be limited. 

### LINKED ISSUE #24236: When Experiment is running Variant modifications should be limited.
When the experiment is running: 

- Variants only can be viewed, change the `edit` button for `view`
- Variants Name can't be edited, remove the ability to be edited.

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Variants only can be viewed, change the edit button for view
* Variants Name can't be edited, remove the ability to be edited.

### Screenshots

**Before**
image

**After**
image

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.ts [hunk 1]
```diff
@@ -166,7 +167,13 @@ we want to show the lock off so the new user can steal the lock
...
     private getStateModeOptions(pageState: DotPageRenderState): SelectItem[] {
         const items = this.variant
-            ? [...(!this.variant.variant.isOriginal ? ['edit'] : []), 'preview']
+            ? [
+                  ...(!this.variant.variant.isOriginal &&
+                  this.variant.experimentStatus === DotExperimentStatusList.DRAFT
+                      ? ['edit']
+                      : []),
+                  'preview'
+              ]
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.ts:176]: Nested ternary are hard to understand... let's keep it simple, if are your friend too.

You might want to extract this condition to a function with a meaninful name.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.html [hunk 2]
```diff
@@ -24,7 +24,10 @@ <h2 class="flex align-items-center gap-1 uppercase">
                 <div class="flex flex-row align-items-center align-content-between gap-2">
                     <div class="title flex flex-grow-1 flex-row align-items-center gap-2">
                         <ng-container
-                            *ngIf="variant.name === defaultVariantName; else editVariantNameTpl"
+                            *ngIf="
+                                variant.name === defaultVariantName || !vm.isExperimentADraft;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.html:28]: This looks like a very specific condition that you want to put in the store.

For example `isExperimentADraft` means that there are other state, that should be reflected in the state.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.ts (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.ts:176]: I agree.

**Review by oidacra — APPROVED** [review]
Only the conditions mentioned by @fmontes

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.html (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.html:28]: the other condition in particular refers to condition of each of the variants that can have the experiment that can be 1 to 3, is dynamic.

**Review by oidacra — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.spec.ts [hunk 3]
```diff
@@ -203,6 +236,66 @@ describe('DotEditPageStateControllerComponent', () => {
...
+            });
+
+            it('should show only the preview tab when experiment is not Draft', async () => {
+                componentHost.variant = {
+                    ...dotVariantDataMock,
+                    experimentStatus: DotExperimentStatusList.RUNNING
+                };
+                fixtureHost.detectChanges();
+
+                const selectButton = de.query(By.css('p-selectButton')).componentInstance;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.spec.ts:265]: testId

**Review by fmontes — APPROVED** [review]

**Pull request merged**